src/clients/dotnet/TigerBeetle/AssertionException.cs

```diff
@@ -2,6 +2,12 @@
 
 namespace TigerBeetle
 {
+    /// <summary>
+    /// The exception that is thrown when an assertion used for correctness checks is triggered.
+    /// Correctness check assertions differ from other assertions offered by the .Net ecosystem
+    /// i.e. Tracer.Assert and Debug.Assert, because they are not meant to be disabled or re-routed.
+    /// It's recommended that the application handle AssertionException as unrecoverable fatal errors.
+    /// </remarks>
```

Comment:
Opening tag (summary) and closing tag (remarks) do not match?

Reply:
Nice catch, surely they don't match 😳

Important thing, this exposed that we are not generating the XML documentation file. Otherwise, it should be an error (or maybe a warning) during the build. I'll tackle that in a follow-up PR.